### PR TITLE
Setup helix ilc runs

### PR DESCRIPTION
**Pull request by joperezr**
cc: @MattGal @weshaggard 

This changes will setup the payloads required to execute tests on Helix. I am also commenting out the actual test execution for now, in order to reduce noise and only get results of ilc compilation.

### CODE REVIEW

File: Tools-Override/CloudTest.targets [hunk 1]
```diff
@@ -428,10 +428,25 @@
       </ForUpload>
     </ItemGroup>
   </Target>
+  
+  <Target Name="CopyTestILCToRuntimeDir" Condition="'$(BuildingUAPAOTVertical)' == 'true'">
```
Comment (MattGal) [inline Tools-Override/CloudTest.targets]: Why is this needed?  If it's a Helix run, we can just zip up the path without having to copy it recursively.  If it's a local run, you can run it from where ever it already is copied..

Comment (joperezr) [inline Tools-Override/CloudTest.targets]: this target will only come into play if you invoke /t:CloudBuild on tests.builds so if you are running locally it will never execute, meaning that you will still use the ilc from wherever you had it locally. If you do invoke CloudBuild, then we copy it so that it goes with the shared payload.

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test please (#16430)

File: Tools-Override/CloudTest.targets (hunk 1 above)
Comment (weshaggard) [inline Tools-Override/CloudTest.targets]: Yes why can't we just setup the location without needing to do the copy?

File: Tools-Override/tests.targets [hunk 2]
```diff
@@ -199,15 +199,20 @@
 
     <PropertyGroup Condition="'$(UseDotNetNativeToolchain)' == 'true'">
       <ILCBuildType Condition="'$(ILCBuildType)' == ''">ret</ILCBuildType>
+      <_TestILCFolder>$(TestILCFolder)</_TestILCFolder>
+      <!--When archiving tests, ilc.exe will be located at %RUNTIME_PATH%\..\TestILC-->
+      <_TestILCFolder Condition="'$(ArchiveTests)' == 'true'">%RUNTIME_PATH%\..\TestILC</_TestILCFolder>
+      <!--Temporarily comment out the test command line when executing on Helix so that we can analyze how many projects fail ilc compilation-->
+      <TestCommandLine Condition="'$(ArchiveTests)' == 'true'">:: $(TestCommandLine)</TestCommandLine>
```
Comment (weshaggard) [inline Tools-Override/tests.targets]: For testing did you want to change this to "echo" so we can more easily see what it would have done in the logs?

File: Tools-Override/tests.targets [hunk 3]
```diff
@@ -199,15 +199,20 @@
 
     <PropertyGroup Condition="'$(UseDotNetNativeToolchain)' == 'true'">
       <ILCBuildType Condition="'$(ILCBuildType)' == ''">ret</ILCBuildType>
+      <_TestILCFolder>$(TestILCFolder)</_TestILCFolder>
+      <!--When archiving tests, ilc.exe will be located at %RUNTIME_PATH%\..\TestILC-->
+      <_TestILCFolder Condition="'$(ArchiveTests)' == 'true'">%RUNTIME_PATH%\..\TestILC</_TestILCFolder>
```
Comment (weshaggard) [inline Tools-Override/tests.targets]: I think it is a bad-idea to assume that the runtime payload will end up putting TestILC in the same relative path. We need to be sure that the the full payload will have the correct structure.

**Comment by joperezr** [comment]
Closing for now since we only really needed this to produce this list: #16764
Will need to reopen and reiterate once we are done with that issue.
